enhancements/baremetal/coreos-image-in-release.md

```diff
@@ -0,0 +1,269 @@
+should now be viable after [fixes to the
+installer](https://github.com/openshift/installer/pull/4413).
+
+The correct version of CoreOS to use is available from the [stream
+metadata](https://github.com/openshift/installer/blob/master/data/data/rhcos-stream.json)
+in the openshift-installer. This could be obtained by using the installer
+container image as a build image. Deriving the container image from the
+installer container has the convenient side-effect of always triggering a
+rebuild when the version changes.
+
+Because OpenShift container images are built in an offline environment, it is
+not possible to simply download the image from the public Internet at container
+build time. Instead this will be accomplished by *MAGIC*. Uploading the ISOs to
+the lookaside cache in Brew is one possibility.
+
+Different processor architectures require different ISOs (though all are
+represented in the stream metadata). It may make sense to have a separate
+Dockerfile for each architecture. For the foreseeable future, we only need to
+be able to pull a container for the local architecture that contains an ISO for
+that same architecture. In the future, it is possible that the containers for
+each architecture's ISO may themselves need to be multi-architecture, so that
+it is possible to run a container image containing an ISO for a *different*
+architecture.
+
+The RHCOS release pipeline may need to be adjusted to ensure that the latest
+ISO is always available in the necessary location.
+
+OKD uses Fedora CoreOS images instead of RHEL CoreOS, but the URLs are
+nevertheless listed in the installer's [stream
+metadata](https://github.com/openshift/installer/blob/master/data/data/fcos-stream.json).
+The build environment for OKD is *not* disconnected, so the container build can
+download the image directly.
+
+### Risks and Mitigations
+
+If the build pipeline does not guarantee the latest RHCOS version gets built
+into the container image, then baremetal platform users may miss out on the
+latest bug and security fixes.
+
+## Design Details
+
+### Open Questions [optional]
+
+How will we provide up-to-date images to the container build?
+
+### Test Plan
+
+The expected SHA256 hashes of the ISO and PXE files are available metadata in
+the cluster, so we should be able to verify at runtime that we have the correct
+image.
+
+### Graduation Criteria
+
+#### Dev Preview -> Tech Preview
+
+N/A
+
+#### Tech Preview -> GA
+
+N/A
+
+#### Removing a deprecated feature
+
+N/A
+
+### Upgrade / Downgrade Strategy
+
+The container registry will always contain an image with latest ISO. This will
+get rolled out to the `image-customization-controller` pod, and future boots of
+the deploy image will be based on the new ISO. The restart of ironic during the
+upgrade should ensure that any BaremetalHosts currently booted into the deploy
+image (i.e. not provisioned) will be rebooted into the new one.
```

Comment:
do these pods run on the host cluster's control-plane nodes?  Because you can set compute MachineConfigPools `paused: true` and they won't be rolled during the update or block the update's completion.  And even if your pool is unpaused, the machine-config operator may not drain and reboot nodes if the `machine-os-content` image (holding RHCOS) didn't get bumped by the update (most patch updates bump it, but not all).  I'm not sure where the ironic pod is from, but I'd guess that may or may not get bumped in updates that happen to bump the boot image.  So unless you're running the pod off the boot image itself with a by-digest pullspec, I think expecting something to come around and roll you may be overly optimistic.

Reply:
> Does the CBO need to keep track of the image reference and ensure container that use the image are told when it changes?

It does already.